pkg/occlient/occlient.go

```diff
@@ -1056,32 +1056,50 @@ func (c *Client) FollowBuildLog(buildName string) error {
 // kind is always required (can be set to 'all')
 // name can be omitted if labels are set, in that case set name to ''
 // if you want to delete object just by its name set labels to nil
-func (c *Client) Delete(kind string, name string, labels map[string]string) (string, error) {
+func (c *Client) Delete(labels map[string]string) error {
+	// convert labels to selector
+	selector := util.ConvertLabelsToSelector(labels)
 
-	args := []string{
-		"delete",
-		kind,
+	// Delete DeploymentConfig
+	err := c.appsClient.DeploymentConfigs(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete deploymentconfig")
```

Comment:
If for some reason deletion of one Kind fails it should still continue and try to delete the others.

For example, if there was an error during DeploymentConfig deletion it should still continue and try to delete Route .... 

At the end error should be returned with the list all errors that occurred.